src/ng/compile.js

```diff
@@ -2592,11 +2587,13 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '=':
-            if (optional && !attrs[attrName]) {
-              return;
+            if (!hasOwnProperty.call(attrs, attrName)) {
```

Comment:
@caitp - in https://github.com/angular/angular.js/issues/12144#issuecomment-119284219 you mentioned some additional code required for Angular Material. 
*  Is that still true?
*  If yes, should we modify the Angular Material directives with optional 2-way binding?